pkg/labels/consul_applicator.go

```diff
@@ -201,9 +201,23 @@ func (c *consulApplicator) mutateLabels(labelType Type, id string, labels map[st
 	return nil
 }
 
-// TODO: replace mutateLabels() with this transaction-using implementation
-func (c *consulApplicator) mutateLabelsTxn(ctx context.Context, labelType Type, id string, labels map[string]*string) error {
-	l, index, err := c.GetLabelsWithIndex(labelType, id)
+type LabelFetcher interface {
+	GetLabelsWithIndex(labelType Type, id string) (Labeled, uint64, error)
+}
+
+// mutateLabelsTxn adds operations to the transaction within the passed context to safely
+// make the label mutations requested. It's written as a package level function to avoid
```

Comment:
I've gotten confused with what you mean by package level function. I have inferred that it does not have anything to do with exported-ness because `mutateLabelsTxn` is not exported, but `RemoveLabelTxn` is, and both are claimed to be package level. Can you help me